back/src/Guide/Main.hs

```diff
@@ -122,34 +121,47 @@ lucidWithConfig x = do
 -- The entry point
 ----------------------------------------------------------------------------
 
--- | Start the site.
+-- | Backend uses command line interface:
+{-
+To see help run command:
+$ guide --help
+
+Guide is wiki
+
+Usage: guide [-v|--version] [COMMAND]
+
+
+Available options:
+  -h,--help                Show this help text
+  -v,--version             Show Guide version
+
+Available commands:
+  run                      Run server
+  dry-run                  Load database and exit
+  load-public              Load PublicDB, create base on it and exit
+
+NOTE:
+Command 'guide' is the same as 'guide run'
+
+-}
+
+-- | Backend runner.
 main :: IO ()
-main = mainWith =<< readConfig
-
--- | Start the site with a specific 'Config'.
-mainWith :: Config -> IO ()
-mainWith config@Config{..} = withLogger config $ \logger -> do
-  args <- getArgs
-  let option = headDef "" args
-  when (option == "--dry-run") $ do
-    db :: DB <- openLocalStateFrom "state/" (error "couldn't load state")
-    logDebugIO logger "loaded the database successfully"
-    closeAcidState db
-    exitSuccess
-  -- USAGE: --load-public <filename>
-  -- loads PublicDB from <filename>, converts it to GlobalState, saves & exits
-  when (option == "--load-public") $ do
-    let path = fromMaybe
-          (error "you haven't provided public DB file name")
-          (args ^? ix 1)
-    (Cereal.runGet SafeCopy.safeGet <$> BS.readFile path) >>= \case
-      Left err -> error err
-      Right publicDB -> do
-        db <- openLocalStateFrom "state/" emptyState
-        Acid.update db (ImportPublicDB publicDB)
-        createCheckpointAndClose' db
-        logDebugIO logger "PublicDB imported to GlobalState"
-        exitSuccess
+main = do
+  command <- parseCli
+  config <- readConfig
+  runCommand config command
+
+-- | Choose command from command line.
+runCommand :: Config -> Command -> IO ()
+runCommand config = \case
+  Guide -> runGuide config
+  DryRun -> dryRun config
+  LoadPublic path -> loadPublic path config
+
+-- | Start the site.
+runGuide :: Config -> IO ()
```

Comment:
Yeah, this should definitely be called `runServer` or `serve` or something, "run Guide" seems too abstract.